packages/compiler/test/server/get-hover.test.ts

```diff
@@ -0,0 +1,374 @@
+import { deepStrictEqual } from "assert";
+import { describe, it } from "vitest";
+import { Hover, MarkupKind } from "vscode-languageserver/node.js";
+import { createTestServerHost, extractCursor } from "../../src/testing/test-server-host.js";
+
+describe("compiler: server: on hover", () => {
+  describe("get hover for scalar", () => {
+    it("string scalar", async () => {
+      const hover = await getHoverAtCursor(
+        `
+          scalar Str┆ing;
+        `
+      );
+      deepStrictEqual(hover, {
+        contents: {
+          kind: MarkupKind.Markdown,
+          value: "```typespec\n" + "scalar String\n" + "```",
+        },
+      });
+    });
+
+    it("int16 scalar", async () => {
+      const hover = await getHoverAtCursor(
+        `
+          scalar Int┆16;
+        `
+      );
+      deepStrictEqual(hover, {
+        contents: {
+          kind: MarkupKind.Markdown,
+          value: "```typespec\n" + "scalar Int16\n" + "```",
+        },
+      });
+    });
+  });
+
+  describe("get hover for enum", () => {
+    it("normal enum", async () => {
+      const hover = await getHoverAtCursor(
+        `
+        enum Direc┆tion {
+          North,
+          East,
+          South,
+          West,
+        }
+        `
+      );
+      deepStrictEqual(hover, {
+        contents: {
+          kind: MarkupKind.Markdown,
+          value: "```typespec\n" + "enum Direction\n" + "```",
+        },
+      });
+    });
+
+    it("normal enum member in namespace", async () => {
+      const hover = await getHoverAtCursor(
+        `
+        namespace TestNS;
+        enum Direction {
+          Nor┆th,
+          East,
+          South,
+          West,
+        }
+        `
+      );
+      deepStrictEqual(hover, {
+        contents: {
+          kind: MarkupKind.Markdown,
+          value: "(enum member)\n" + "```typespec\n" + "TestNS.Direction.North\n" + "```",
+        },
+      });
+    });
+  });
+
+  describe("get hover for alias", () => {
```

Comment:
same in general I think having 2 test for each type that are declaration and reference would be nice

Reply:
Updated the tests accordingly. please help to review. thx.